tests/devapps/MSI/HelperService/MSIHelperService/Controllers/GetMSITokenController.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.AspNetCore.Mvc;
+using MSIHelperService.Helper;
+using Swashbuckle.AspNetCore.Annotations;
+
+namespace MSIHelperService.Controllers
+{
+    [ApiController]
+    [Route("[controller]")]
+    [SwaggerTag(description: "Gets MSI Token")]
+    public class GetMSITokenController : ControllerBase
+    {
+        private readonly ILogger _logger;
+
+        /// <summary>
+        /// GetMSITokenController ctor
+        /// </summary>
+        /// <param name="logger"></param>
+        public GetMSITokenController(ILogger<GetMSITokenController> logger)
+        {
+            _logger = logger;
+        }
+
+        /// <summary>
+        /// Gets the Managed Service Identity Token of an Azure Resource
+        /// </summary>
+        /// <param name="identityHeader">IDENTITY_HEADER of the MSI Endpoint</param>
+        /// <param name="uri">URI of the MSI Endpoint</param>
+        /// <param name="azureResource">Resource for which you need the MSI Token</param>
+        /// <returns>
+        /// Returns the MSI token for an azure resource
+        /// </returns>
+        [ProducesResponseType(typeof(ManagedIdentityResponse), (int)MSIHelper.HTTPErrorResponseCode.Status200OK)]
+        [HttpGet]
+        [SwaggerResponse(200, "Returns an Azure Resource MSI Token Response", Type = typeof(ManagedIdentityResponse))]
+        [SwaggerResponse(400, "Returns the error object for any validation failures", Type = typeof(ManagedIdentityResponse))]
+        [SwaggerResponse(500, "Returns the error object for any Server Errors", Type = typeof(ManagedIdentityResponse))]
+        public async Task<ActionResult?> GetRemoteHttpResponse(
```

Comment:
I am not sure how MSAL tests would be able to use this, because MSAL tests expect HttpResponseMessage and not `ManagedIdentityResponse`. 

Not blocking but would like to see a test.

Reply:
i.e. in my mind the interface would be more general purpose

`GetRemoteHttpResponse(string uri, HttpMethod method, string body_for_post, Dictionary<string, string> requestHeaders)`